OJS.Workers.Executors/RestrictedProcessExecutor.cs

```diff
@@ -5,62 +5,40 @@ namespace OJS.Workers.Executors
     using OJS.Workers.Executors.Process;
 
-    public class RestrictedProcessExecutor : IExecutor
+    public class RestrictedProcessExecutor : ProcessExecutor
     {
-        private const int TimeIntervalBetweenTwoMemoryConsumptionRequests = 45;
         private const int TimeBeforeClosingOutputStreams = 300;
-        private const int MinimumMemoryLimitInBytes = 5 * 1024 * 1024;
 
         private static ILog logger;
 
-        private readonly int baseTimeUsed;
-        private readonly int baseMemoryUsed;
+        public RestrictedProcessExecutor(int baseTimeUsed, int baseMemoryUsed, ITasksService tasksService)
+            : base(baseTimeUsed, baseMemoryUsed, tasksService)
+            => logger = LogManager.GetLogger(typeof(RestrictedProcessExecutor));
 
-        public RestrictedProcessExecutor() =>
-            logger = LogManager.GetLogger(typeof(RestrictedProcessExecutor));
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="RestrictedProcessExecutor"/> class with base time and memory used.
-        /// </summary>
-        /// <param name="baseTimeUsed">The base time in milliseconds added to the time limit when executing.</param>
-        /// <param name="baseMemoryUsed">The base memory in bytes added to the memory limit when executing.</param>
-        public RestrictedProcessExecutor(int baseTimeUsed, int baseMemoryUsed)
-            : this()
-        {
-            this.baseTimeUsed = baseTimeUsed;
-            this.baseMemoryUsed = baseMemoryUsed;
-        }
-
-        public ProcessExecutionResult Execute(
+        protected override ProcessExecutionResult InternalExecute(
             string fileName,
             string inputData,
             int timeLimit,
             int memoryLimit,
-            IEnumerable<string> executionArguments = null,
-            string workingDirectory = null,
-            bool useProcessTime = false,
-            bool useSystemEncoding = false,
-            bool dependOnExitCodeForRunTimeError = false,
-            double timeoutMultiplier = 1.5)
+            IEnumerable<string> executionArguments,
+            string workingDirectory,
+            bool useSystemEncoding,
+            double timeoutMultiplier)
         {
-            this.AdjustTimeAndMemoryLimits(ref timeLimit, ref memoryLimit);
-
             var result = new ProcessExecutionResult { Type = ProcessExecutionResultType.Success };
-            if (workingDirectory == null)
-            {
-                workingDirectory = new FileInfo(fileName).DirectoryName;
-            }
 
-            using (var restrictedProcess = new RestrictedProcess(fileName, workingDirectory, executionArguments, Math.Max(4096, (inputData.Length * 2) + 4), useSystemEncoding))
+            using (var restrictedProcess = new RestrictedProcess(
+                fileName,
+                workingDirectory,
+                executionArguments,
+                Math.Max(4096, (inputData.Length * 2) + 4),
```

Comment:
Consider using named variable to suggest its seems odd. Also consider using constants.